### ISSUE TITLE
CPU count calculation and PiNode auto-start

### ISSUE THREAD

**Issue by muratyurdakul75** [pi-node/instructions#411]
Hello
I will have a few suggestions regarding the PiNode application. This suggestion will actually prevent users running Node from being victimized. Therefore, I would be happy if you take these suggestions into consideration in the new version.
 
1. System CPU calculation is incorrect
The PiNode application sends information to your servers at 1-minute intervals. This information includes information such as the number of CPUs and the amount of RAM in the system. However, there is something you are ignoring. In server systems with 2 CPUs, you send the processor number of a single CPU. Naturally, you do not take half of the system into account. For example, my computer has 2 Xeon E5 2698 V4 processors. Each has 40 core processors. A total of 80 cores. However, you send half of this. In addition, WSL uses all 80 cores. This is now possible with the WSL 2.4.5 update. And the PiNode application ignores this. In addition; even if some users have 20 core processors in their system, they can limit this in the wslconfig file and reduce it to 10 processors. Naturally, WSL actually works with 10 processors. However, since you look at the number of system CPUs, you accept the 20-core processor. Naturally, such users will also receive an unfair extra bonus. When calculating the number of processors, it would be more fair to look at the number of CPUs returned from the "docker info" command. This will prevent users running 2 CPUs from being victimized or users who limit the number of processors in the wslconfig file from gaining unfair profits. I kindly ask you to take this into consideration in the new version.
 
2. Automatic start of PiNode application
When Windows first starts, the PiNode application is automatically started. This is a good thing for most users. However, starting automatically does not allow the PiNode application to run in admin mode. I and some users like me prefer to start the PiNode application automatically from the Task Scheduler environment of Windows. Because when started from here, we can start the application in admin mode. However, since the PiNode application starts the PiNode application when Windows first starts, the Task Scheduler does not work for the application to run. You can turn off the automatic start of the application via Task Manager. However, when the PiNode application runs, it sets itself to start automatically again. Naturally, the change I made via Task Manager does not work. My suggestion; It would be better to add the automatic start feature of the PiNode application as a setting on the Profile page. The user who wants to can start it automatically, and the user who wants to can turn off this feature.

Please consider these suggestions. The items I suggest (especially item 1) are important for running a healthier Node. Thank you very much for your efforts.

**Comment by muratyurdakul75** [comment]
Additionally; the Docker CPU usage rate you send at 1 minute intervals also gives incorrect information. Because that rate is not calculated based on 100%. It is calculated based on 100% * WSL CPU count. Like 2000% or 4000%. Therefore, in order to calculate the Docker CPU usage rate fairly for all users, you need to look at the CPU count in the "docker info" command. For this reason, the calculation is not fair at this stage.

**Comment by muratyurdakul75** [comment]
I hope the CT team has seen this issue. Because it is an important issue. This correction needs to be made to prevent users from being treated unfairly or to prevent those who win bonuses unfairly.

**Comment by allen960** [comment]
To whom it may concern:

As observed by Mr Murat in his comments above, the number of cpu cores / threads should be picked up from the Docker. 

Presently, the pre-release WSL version 2.4.8 is already able to get the total number of threads attributed to the computer system's CPU (or CPUs if the system has two sockets populated with CPUs), and this number is passed on to the Docker. As such, the Pi Network app should obtain the number of cores/threads from the Docker instead of just taking the number attributed to only one CPU on the system. There are Pioneers who use dual-socket computer to run their nodes. As these Pioneers have invested in such a system to support the Pi Network, it would only be fair for them to be credited for the total number of cores/threads of the two CPUs that they are contributing to the network. I look forward to seeing this implementation in the next version of Pi Network.

Thank you!

**Comment by allen960** [comment]
It seems that the latest release of pi network app has fixed the problem on cpu cores/threads count. Thank you!

**Comment by Heisenberg2nd** [comment]
Since I should change my PC, among the non-workstation CPUs which would be the best choice? I would say the 14900k for the number of CPUs.. correct me if I'm wrong

**Comment by Ziggy644** [comment]
When I look in the HTTP traffic of the pi app, I found a JSON response that calculates the node reward. However, In my case, I also have a dual Xeon e5-2698v4 setup. (40 cores, 80 threads) But in the latest version the app still shows 32 cores used...

[pi node.json](https://github.com/user-attachments/files/24891827/pi.node.json)

**Comment by Ziggy644** [comment]
It's a limitation provided by CT. CT limits 'rewarded' cores to 32. I found out because the stats, sent to the pi servers contain the correct cpu core amount.

**Comment by allen960** [comment]
Does docker show the correct number of CPU for you?

**Comment by MurattiIng** [comment]
> It's a limitation provided by CT. CT limits 'rewarded' cores to 32. I found out because the stats, sent to the pi servers contain the correct cpu core amount.

If they're really ignoring processors with more than 32 cores, that's extremely unfortunate. It's unfair. I hope they address this issue.

**Comment by Ziggy644** [comment]
> Does docker show the correct number of CPU for you?
> […](#)

Yes Docker Reports all 80 cores are in use

**Comment by MurattiIng** [comment]
> > Does docker show the correct number of CPU for you?
> > […](#)
> 
> Yes Docker Reports all 80 cores are in use

Yes. I see 256 processors in Docker. But the fact that only 32 of them are accepted feels unfair to the effort.

**Comment by Ziggy644** [comment]
<img width="740" height="342" alt="Image" src="https://github.com/user-attachments/assets/3f6f0e9f-2226-4a4d-9489-b24ab5f98344" />
